hydroflow_lang/src/graph/ops/join_multiset.rs

```diff
@@ -0,0 +1,69 @@
+use syn::{parse_quote, parse_quote_spanned};
+
+use super::{
+    FlowProperties, FlowPropertyVal, OperatorCategory, OperatorConstraints, WriteContextArgs,
+    RANGE_0, RANGE_1,
+};
+use crate::graph::{OpInstGenerics, OperatorInstance};
+
+/// > 2 input streams of type <(K, V1)> and <(K, V2)>, 1 output stream of type <(K, (V1, V2))>
+///
+/// This operator is equivalent to `join` except that the LHS and RHS are collected into multisets rather than sets before joining.
```

Comment:
hmm order is preserved I think so multisets might not be the right word

Reply:
Given that there are many (multiset) join algorithms, that seems like a side-effect of the implementation that we may not want to provide as a guarantee, so I wouldn't sweat it.